homeassistant/components/plex/server.py

```diff
@@ -39,12 +43,34 @@
 plexapi.X_PLEX_VERSION = X_PLEX_VERSION
 
 
+def debounce(func):
+    """Decorate function to debounce callbacks from Plex websocket."""
+
+    @callback
+    def call_later_listener(self, _):
+        """Handle call_later callback."""
+        self.debounce = None
+        self.hass.async_add_executor_job(func, self)
+
+    @wraps(func)
+    def wrapper(self):
+        """Schedule async callback."""
+        if self.debounce:
+            self.debounce()
+        remove_listener = async_call_later(
+            self.hass, DEBOUNCE_TIMEOUT, partial(call_later_listener, self),
```

Comment:
Home Assistant internal clock runs at 1s, so 0.5s is not possible. If you want that granularity you should consider using `hass.loop.call_later`